### DISCUSSION TITLE
Binary Sensors

### DISCUSSION THREAD

**Discussion by wernerhp** [hawky358/hass_ids_hyyp#21] [General]
Heya, is there any way to get the binary sensor states when the system is disarmed?

**Comment by hawky358** [comment]
Hello, 

Yes but the updates are limited to once every 30 seconds, so the state may be missed if it isn't persistent for 30 seconds.

See the [readme](https://github.com/hawky358/hass_ids_hyyp#features)

Under `switch.[zone_name]` and sub section `violated` attribute where it's explained.

**Comment by wernerhp** [comment]
How do I get it to persist the state for 30 seconds?

**Comment by hawky358** [comment]
What this means is that the zone itself needs to be "violated" for around 30s to be 100% sure. I.e. The state needs to persist for 30s.

E.g. An open door.

The alarm panel isn't currently reporting violated zones to ids server as a push. You only get this state when requested.

This is just the way the panel itself is designed /programmed.

**Comment by wernerhp** [comment]
Does that mean it doesn't work for a PIR, since it's only violated for a moment and not >30s?

A PIR does trigger the alarm and the notification presumably comes through even if it's disarmed within 30s?

**Comment by hawky358** [comment]
It depends on what you mean by "notification".

The sensors themselves are only updated once every 30 seconds*.  Anything that happens between polls may be missed if the state reverts back to the original state before the next poll.  It's just the way the IDS server set up works, it doesn't push information to the clients, the clients must poll. Normally your phone polls when you open the app.

*When performing actions, an immediate update is queued by the integration e.g. arm/disarm/bypass, so you'll get an update within 5 or so seconds.

On version 1.5.0+ I've implemented "push notifications". This uses the same system that IDS uses to send your phone the push notifications which are registered by the app.
This way you'll receive instant notifications via the `sensor.[site]_ids_push_notifications` sensor. Note it will only send you the notifications that IDS will normally send you e.g. trigger, arm, disarm, bypass, etc. You won't get "violation of disarmed zones" via the IDS push notification service, because IDS doesn't get this information from the panel unless polled nor do they push the information to clients.

You can look at the [Readme](https://github.com/hawky358/hass_ids_hyyp/tree/1.5.2#features) regarding the notification sensor, but it's basically as I explained it above.
